src/odemis/gui/cont/views.py

```diff
@@ -325,31 +404,55 @@ def _on_chamber_state(self, state):
             # sample have probably nothing in common with this new sample
             self._data_model.stage_history.value = self._data_model.stage_history.value[-1:]
 
+    def _update_ovv(self):
+        """ Update the overview image with the currently active stream. """
+        if self._data_model.streams.value:
+            for s in self._data_model.streams.value:
+                if s.is_active.value and s.image.value is not None:
+                    img = s.image.value
+                    if isinstance(s, acqstream.OpticalStream):
+                        opt = img
+                        sem = None
+                    if isinstance(s, acqstream.EMStream):
+                        sem = img
+                        opt = None
+
+                    # Update optical or sem image, depending on which stream is active
+                    if opt is not None:
+                        self.im_opt = insert_tile_to_image(opt, self.im_opt)
+                    if sem is not None:
+                        self.im_sem = insert_tile_to_image(sem, self.im_sem)
+
+                    # Merge optical and sem overview images
+                    self.ovv_im = merge_opt_sem(self.im_opt, self.im_sem)
+
+                    # Update display
+                    if self.upd_stream:
```

Comment:
Why would `self.upd_stream` be equivalent to False? It seems this `if` block is always executed.

Moreover, I'd recommend to only run it once per update (at most). Maybe you could just check `.curr_s` instead of looking at every stream (which should be doing the same thing as long as not 2 streams are playing simultaneously). Or you'd need to check if ovv_im has been updated.